src/iop/colorbalancergb.c

```diff
@@ -1414,6 +1414,30 @@ void color_picker_apply(dt_iop_module_t *self, GtkWidget *picker, dt_dev_pixelpi
 }
 
 
+static void _YchToRGB(dt_aligned_pixel_t *RGB_out, float chroma, float hue,
```

Comment:
Just noticed that now, why not declare this RGB only, then remove the declaration of RGB below and the final `memcpy()`. This will gain some CPU cycles.

Reply:
Done and added the else to the conditional that check for non_linearlut as we would otherwise have left RGB_out unset

Reply:
Which interestingly was likely an oversight in the original code as RGB_linear was never copied into RGB in case output_profile->nonlinearlut wasn't true thus leaving RGB initilized to { 0. }

Reply:
> Which interestingly was likely an oversight in the original code as RGB_linear was never copied into RGB in case output_profile->nonlinearlut wasn't true thus leaving RGB initilized to { 0. }

Indeed, good catch.